File: src/main/java/me/zeroeightsix/kami/command/commands/VanishCommand.java
package me.zeroeightsix.kami.command.commands;

import me.zeroeightsix.kami.command.Command;
import net.minecraft.entity.Entity;

import static me.zeroeightsix.kami.util.MessageSendHelper.sendChatMessage;

/**
 * Created by d1gress/Qther on 25/11/2017.
 */

public class VanishCommand extends Command {

    private static Entity vehicle;

    public VanishCommand() {
        super("vanish", null);
        setDescription("Allows you to vanish using an entity");
    }

    @Override
    public void call(String[] args) {
        if (mc.player.getRidingEntity() != null && vehicle == null) {
            vehicle = mc.player.getRidingEntity();
            mc.player.dismountRidingEntity();
            mc.world.removeEntityFromWorld(vehicle.getEntityId());
            sendChatMessage("Vehicle " + vehicle.getName() + " removed.");
        } else {
            if (vehicle != null) {
                vehicle.isDead = false;
                mc.world.addEntityToWorld(vehicle.getEntityId(), vehicle);
                mc.player.startRiding(vehicle, true);
                sendChatMessage("Vehicle " + vehicle.getName() + " created.");
                vehicle = null;
            } else {
                sendChatMessage("No Vehicle.");
            }
        }
    }
}

File: src/main/kotlin/com/lambda/client/command/commands/VanishCommand.kt
package com.lambda.client.command.commands

import com.lambda.client.command.ClientCommand
import com.lambda.client.util.text.MessageSendHelper.sendChatMessage
import com.lambda.client.util.text.formatValue
import net.minecraft.entity.Entity

object VanishCommand : ClientCommand(
    name = "vanish",
    description = "Allows you to vanish using an entity."
) {
    private var vehicle: Entity? = null

    init {
        executeSafe {
            if (player.ridingEntity != null && vehicle == null) {
                vehicle = player.ridingEntity?.also {
                    player.dismountRidingEntity()
                    world.removeEntityFromWorld(it.entityId)
                    sendChatMessage("Vehicle " + formatValue(it.name) + " removed")
                }
            } else {
                vehicle?.let {
                    it.isDead = false
                    world.addEntityToWorld(it.entityId, it)
                    player.startRiding(it, true)
                    sendChatMessage("Vehicle " + formatValue(it.name) + " created")
                    vehicle = null
                } ?: sendChatMessage("Not riding any vehicles")
            }
        }
    }
}